Added authorization to backend endpoints

Mo-sissoko opened the pull request:
### ℹ️ Issue

Closes #25

### 📝 Description

- Added authorization to the yaml template for /forms, /forms{id}/notes, and /forms{id} endpoints.
- Added cognito authorization to the yaml temlate

### ✔️ Verification

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves. 

Provide screenshots of any new components, styling changes, or pages. 
1. Change directories over to backend
2. npm run build
3. sam deploy --guided --capabilities CAPABILITY_NAMED_IAM

Once it's deployed,  /forms, /forms{id}/notes, and /forms{id} endpoints should be protected and require authorization.

![image](https://user-images.githubusercontent.com/73724045/232352119-b51c18f6-22ea-4ffc-b415-9e075e47a428.png)
![image](https://user-images.githubusercontent.com/73724045/232352138-8b2fac74-9653-43ec-a98c-42995ba91473.png)
![image](https://user-images.githubusercontent.com/73724045/232352178-923472a6-fdf7-4acc-bbf0-891fec3e049b.png)

### 🏕️ (Optional) Future Work / Notes

Did you notice anything ugly during the course of this ticket? Any bugs, design challenges, or unexpected behavior? Write it down so we can clean it up in a future ticket!

Mo-sissoko committed:
Added authorization to backend endpoints

megandouli reviewed (changes requested):
awesome!! I don't want to deploy this and have the entire frontend break... could you update the frontend code as well to send the `Authorization` token in requests to protected endpoints?

Mo-sissoko committed:
Added authorization to request headers

- Added Authorization to allowed headers in CORs
- Update the requests to add auth in them

Mo-sissoko committed:
Merge branch 'main' into protect-backend-endpoints

megandouli reviewed (changes requested):
NVM

megandouli commented on apps/backend/src/handlers/getAllForms.ts:
```diff
@@ -7,6 +7,18 @@ import { validateMethodType } from '../utils/validateMethodType.js';
+      statusCode: 400,
+      headers,
+      body: `getMethod only accepts GET method, you tried: ${event.httpMethod} method.`,
+    };
```
we're missing a closing brace here

megandouli reviewed (changes requested):
wait sorry for the back and forth, I don't think you merged main in properly.

Mo-sissoko committed:
Fixed merging issues.

megandouli commented on apps/backend/src/handlers/getAllForms.ts:
```diff
@@ -1,25 +1,36 @@
-    return response;
+  if (event.httpMethod !== 'GET') {
+    return createResponse(
+      400,
```
can you reuse Nate's refactored helper methods? e.g. `validateMethodType` instead of `if (event.httpMethod !== 'GET')`

megandouli replied:
also use `executeHandler`, take a look at what main looks like right now for how to use Nate's helpers

megandouli commented on apps/frontend/src/utils/sendRequest.ts:
```diff
@@ -9,6 +10,19 @@ import {
 import { AdminNotes, formSchema } from '../types/formSchema';
 
+const authenticatedAxios
+= axios.create();
```
nit: fix the whitespace / unnecessary line break here

Mo-sissoko replied:
Fixed

Mo-sissoko committed:
Update sendRequest.ts

megandouli commented on apps/backend/src/handlers/patchForm.ts:
```diff
@@ -9,6 +9,18 @@ import { validateRequestBody } from '../utils/validateRequestBody.js';
+      statusCode: 400,
+      headers,
+      body: `patchMethod only accepts PATCH method, you tried: ${event.httpMethod} method.`,
+    };
```
also missing a closing `}` here

Mo-sissoko committed:
Used abstracted methods and fixed some minimal issues

megandouli reviewed (approved):
nice!
